### PR TITLE
Added fe1-web documentation

### PR DESCRIPTION
**Pull request by pierluca**
Here's a first batch of documentation.
I'll address the remaining points in a subsequent PR.

I'm still hesitant on whether to have Markdown files in every folder (e.g. `ingestion/README.md`), or a fully JS-based documentation using TypeDoc (e.g., the `index.ts` files).

TypeDoc is quite powerful, but for now it doesn't run due to compilation issues ; furthermore, Markdown files might be more easily accessible on Github.

What do you think?

### CODE REVIEW

File: fe1-web/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+for more information.
+
+All the messages are encoded using JSON and are validated using
+[JSON Schemas](https://json-schema.org) as defined in the
+[protocol folder](https://github.com/dedis/student_21_pop/tree/master/protocol).
+
+[Protocol Specifications](https://docs.google.com/document/d/1fyNWSPzLhM6W9V0VTFf2waMLiJGcscy7wa4bQlLkySM)
+also gives an introduction to the different message formats. However, note that
+the [schemas](https://github.com/dedis/student_21_pop/tree/master/protocol) in this
+repository are **always** the source of truth and are more up to date than the Google Doc.
```
Comment (pierluca) [inline fe1-web/docs/README.md:98]: I've more or less stolen this part from Gaurav, which makes me think it should belong to a dedicated project-wide documentation :)

Comment (gnarula) [reply fe1-web/docs/README.md:98]: I think this much duplication is fine. More links to follow, more chances to get lost/distracted ;)

File: fe1-web/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+#### Managing the application state
+
+In order for the application to perform as a user would expect, it needs to manage
+its own internal state robustly. Its internal state is made up of local data,
+such as a user's cryptographic material and preferences, but also of the local
+representation of the application's view of the whole PoP system and its state.
+
```
Comment (gnarula) [inline fe1-web/docs/README.md:106]: Would be nice to give an overview of how the cryptographic material is stored and used here

File: fe1-web/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+their `fromJSON` method to get an idea about how to implement a new type.
+
+#### UI elements
+
+... TBD ...
+Will mention difference between `components` (UI component library) and `parts` (unique screens).
+
+#### Validation
+
+All the incoming messages are validated using the `network\validation` package,
```
Comment (gnarula) [inline fe1-web/docs/README.md:215]:
```suggestion
All the incoming messages are validated using the `network/validation` package,
```

Comment (pierluca) [reply fe1-web/docs/README.md:215]: Windows user spotted 🤣

File: fe1-web/docs/README.md (hunk 2 above)
Comment (pierluca) [reply fe1-web/docs/README.md:106]: Excellent idea !

**Comment by gnarula** [comment]
I don't know much about TS land but I gather `TSDoc` is the Typescript equivalent to `godoc`/Javadoc right? If so, IMHO  it's better we use it for consistency and the fact that students will get to use something that's standard for documentation.

File: fe1-web/network/NetworkManager.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { NetworkConnection } from './NetworkConnection';
 import { NetworkError } from './NetworkError';
 import { defaultRpcHandler, JsonRpcHandler } from './RpcHandler';
 
+const defPort = 9000;
```
Comment (nkcr) [inline fe1-web/network/NetworkManager.ts:6]: unclear what it means. I read it first as "DefinitionPort".

**Review by nkcr — APPROVED** [review]
Very minor comments. I learnt a lot from the README which is very informative!

File: fe1-web/network/NetworkManager.ts (hunk 4 above)
Comment (pierluca) [reply fe1-web/network/NetworkManager.ts:6]: removed in a later commit

**Comment by pierluca** [comment]
@nkcr can you re-review and merge? thanks :)

**Review by nkcr — APPROVED** [review]

**Pull request merged**